### PR TITLE
Libjulia 1.5.4

### PR DESCRIPTION
**Pull request by barche**
Rebase of PR #2898

### CODE REVIEW

**Comment by barche** [comment]
I think this is good to go, before we merge PR #3411 

CC @fingolfin

File: L/libjulia/common.jl [hunk 1]
```diff
@@ -335,7 +335,11 @@ function build_julia(ARGS, version::VersionNumber)
         error("Unsupported Julia version")
     end
 
-    julia_compat = version ≥ v"1.7" ? "1.6" : "1.0"
+    julia_compat = version ≥ v"1.6" ? "1.6" : "1.0"
+
+    if version == v"1.5.4"
+        julia_compat = "~1.5.4"
```
Comment (fingolfin) [inline L/libjulia/common.jl:341]: Why?

File: L/libjulia/libjulia@1.5.4/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,2 @@
+include("../common.jl")
+build_julia(ARGS, v"1.5.4")
```
Comment (fingolfin) [inline L/libjulia/libjulia@1.5.4/build_tarballs.jl:2]: Why is this a new file, instead of updating the existing 1.5 builder?

**Comment by fingolfin** [comment]
So the ABI of Julia 1.5.4 is incompatible to that of 1.5.0-1.5.3. At least in parts.

So any JLL built using libjulia_jll 1.5.4 and which uses the incompatible portions of the ABI, should ideally have a julia_compat requiring ~1.5.4. AFAIK right now that's only libtask_jll. But that JLL needs to set the compat bound, not libjulia

File: L/libjulia/libjulia@1.5.4/build_tarballs.jl (hunk 2 above)
Comment (barche) [reply L/libjulia/libjulia@1.5.4/build_tarballs.jl:2]: To emphasize the incompatibility with other versions, it is indeed debatable wether this is necessary.

File: L/libjulia/common.jl (hunk 1 above)
Comment (barche) [reply L/libjulia/common.jl]: Indeed not needed, given your explanation below. I removed the 1.5.4 compat.

**Comment by barche** [comment]
Cleaned this up following @fingolfin 's comments, there was indeed no need for a separate 1.5.4 directory.

File: L/libjulia/common.jl [hunk 3]
```diff
@@ -335,7 +335,7 @@ function build_julia(ARGS, version::VersionNumber)
         error("Unsupported Julia version")
     end
 
-    julia_compat = version ≥ v"1.7" ? "1.6" : "1.0"
+    julia_compat = version ≥ v"1.6" ? "1.6" : "1.0"
```
Comment (fingolfin) [inline L/libjulia/common.jl:338]: Why is this needed in this PR? Seems to belong to the next one?

File: L/libjulia/common.jl [hunk 4]
```diff
@@ -286,16 +285,17 @@ function build_julia(ARGS, version::VersionNumber)
         push!(dependencies, Dependency("LibOSXUnwind_jll", compat="0.0.5"))
     elseif version < v"1.7"
         push!(dependencies, Dependency("LibOSXUnwind_jll", compat="0.0.6"))
+        push!(dependencies, Dependency("LibUV_jll"))
     end
 
     if version < v"1.6"
-        push!(dependencies, BuildDependency("SuiteSparse_jll", compat="5.4.0"))
+        push!(dependencies, BuildDependency(PackageSpec(name="SuiteSparse_jll", version="5.4.0")))
```
Comment (fingolfin) [inline L/libjulia/common.jl:292]: Could you briefl say what this change is about?

File: L/libjulia/common.jl [hunk 5]
```diff
@@ -286,16 +285,17 @@ function build_julia(ARGS, version::VersionNumber)
         push!(dependencies, Dependency("LibOSXUnwind_jll", compat="0.0.5"))
     elseif version < v"1.7"
         push!(dependencies, Dependency("LibOSXUnwind_jll", compat="0.0.6"))
+        push!(dependencies, Dependency("LibUV_jll"))
```
Comment (fingolfin) [inline L/libjulia/common.jl:288]: Doesn't this rather belong in your other PR? It seem irrelevant for building 1.5?

File: L/libjulia/common.jl (hunk 3 above)
Comment (barche) [reply L/libjulia/common.jl:338]: Yes, it is not needed here and will also be in the next PR. Doesn't change anything for this build.

File: L/libjulia/common.jl (hunk 4 above)
Comment (barche) [reply L/libjulia/common.jl:292]: `BuildDependency` has no compat keyword, so you need to set it using `PackageSpec`. The original PR didn't touch the `common.jl` file, but it seems the build process uses `common.jl` from master, so I had to fix this here.

File: L/libjulia/common.jl (hunk 5 above)
Comment (barche) [reply L/libjulia/common.jl:288]: This is needed here, because `LibUV` was in the common dependencies but adding it for 1.5.4 makes the build fail on linux i686.

**Comment by barche** [comment]
I think this should be good to merge, right?

File: L/libjulia/common.jl [hunk 6]
```diff
@@ -27,7 +27,7 @@ function build_julia(ARGS, version::VersionNumber)
     checksums = Dict(
         v"1.3.1" => "3d9037d281fb41ad67b443f42d8a8e400b016068d142d6fafce1952253ae93db",
         v"1.4.2" => "76a94e06e68fb99822e0876a37c2ed3873e9061e895ab826fd8c9fc7e2f52795",
-        v"1.5.3" => "be19630383047783d6f314ebe0bf5e3f95f82b0c203606ec636dced405aab1fe",
+        v"1.5.4" => "852122bf1bdefd39307b1dd2aa546e3885d76ede7c07cb04d90814b9510ea9f9",
```
Comment (giordano) [inline L/libjulia/common.jl:30]: I'd add 1.5.4, instead of replacing 1.5.3

File: L/libjulia/libjulia@1.5/build_tarballs.jl [hunk 7]
```diff
@@ -1,2 +1,2 @@
 include("../common.jl")
-build_julia(ARGS, v"1.5.3")
+build_julia(ARGS, v"1.5.4")
```
Comment (giordano) [inline L/libjulia/libjulia@1.5/build_tarballs.jl:2]: One thing is not clear to me: will this be installed for Julia v1.5.0-1.5.3?  Because v1.5.4 broke ABI compatibility with previous versions in the 1.5 series

Comment (barche) [reply L/libjulia/libjulia@1.5/build_tarballs.jl:2]: That's why I initially put 1.5.4 in a separate dir, but as @fingolfin pointed out this package is only used as a build dependency, so it's up to the JLLs using this to set their compat correctly, depending on the version of libjulia they build against.

File: L/libjulia/common.jl (hunk 6 above)
Comment (barche) [reply L/libjulia/common.jl:31]: ok, done!
